recipes/libusb/all/conanfile.py

```diff
@@ -61,31 +61,17 @@ def validate(self):
         if self.options.get_safe("enable_udev"):
             if self.settings.os == "Android":
                 raise ConanInvalidConfiguration("udev can't be enabled for Android yet, since libudev recipe is missing in CCI.")
-            if self.settings.os == "Linux" and self._settings_build.os != "Linux":
```

Comment:
Hey @SpaceIm , you did the changes I'm overriding here. I'm sure you have a good reason to do it this way, so I can revert my changes if needed.
I want Conan to error out if you're cross compiling, since `libudev` will be provided by the OS.
The current logic fails to catch the situation where both, the build OS and the host OS are Linux but you're crosscompiling anyways. E.g. Linux:x86_64 -> Linux:arm
Maybe I'm missing some detail here.

Reply:
By moving to `libudev/system`, `validate()` should be removed in `libusb` recipe and delegated to `libudev` recipe. And obviously `libudev` recipe should raise if cross-building.

Reply:
@ericriff Please follow @SpaceIm review. Otherwise, this PR seems good

Reply:
Oops, I forgot about this PR.
I can do that, but `libudev/system` is not raising when crosscompiling. It only checks fot the OS == Linux
https://github.com/conan-io/conan-center-index/blob/master/recipes/libudev/all/conanfile.py#L17

Reply:
Locally I've updated `libudev/system` to have this check
```
    def validate(self):
        if self.settings.os != "Linux":
            raise ConanInvalidConfiguration("libudev is only supported on Linux.")
        if tools.cross_building(self):
            raise ConanInvalidConfiguration("system packages only supported for native builds")
```
Which doesn't raise when we try to crosscompile `libusb` :thinking:

Reply:
I checked other system packages (`xorg`, `opengl`) and those are not validating that we're not trying to cross build. In all of those it seems that it is expected that the consumer recipe, like this one, does the checking.